src/Serializer/JsonApiSerializer.php

```diff
@@ -72,6 +72,11 @@ public function item($resourceKey, array $data)
 
         unset($resource['data']['attributes']['id']);
 
+        if(isset($resource['data']['attributes']['links'])) {
+            $resource['data']['links'] = $data['links'];
+            unset($resource['data']['attributes']['links']);
+        }
+
         if ($this->shouldIncludeLinks()) {
             $resource['data']['links'] = [
                 'self' => "{$this->baseUrl}/$resourceKey/$id",
```

Comment:
@lukasoppermann 
Are you sure this will work?
I think in this line `$resource['data']['links']` will be fully replaced with the 'self'-link. The custom links will be overwritten.

I think the `JsonApiSerializerTest.php` will succeed anyway because you don't use the `$baseUrl`. But if you use the `$baseUrl` the test will fail.

This should work:
```php
if ($this->shouldIncludeLinks()) {  
  $resource['data']['links'] = array_merge($resource['data']['links'], [
     'self' => "{$this->baseUrl}/$resourceKey/$id",
   ]);
}
```

Correct me if I'm wrong.